packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable i18next/no-literal-string */
+import { 
+  Component, 
+  Element, 
+  Host, 
+  h, 
+  Prop, 
+  forceUpdate,
+  Fragment,
+  Event,
+  EventEmitter
+} from '@stencil/core';
+import i18next from 'i18next';
+
+@Component({
+  tag: 'va-file-input',
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the button input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button
+   */
+  @Prop() buttonText!: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * Allow the input to accept multiple files.
+   */
+  @Prop() multiple?: boolean = false;
+
+  /**
+   * A comma-separated list of unique file type specifiers.
+   */
+  @Prop() accept?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Emit component-library-analytics events on the file input change event.
+   */
+   @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * The event emitted when the file input value changes.
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * file input changes and enableAnalytics is true.
+   */
+   @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.vaChange.emit({files: target.files});
+    /**
+     * Clear the original input, otherwise events will be triggered
+     * with empty file arrays and sometimes uploading a file twice will
+     * not work.
+     */
+    target.value = null;
+
+    if (this.enableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-file-input',
+        action: 'change',
+        details: {
+          label: this.label
+        },
+      });
+    }
+  };
+
+  private handleButtonClick = () => { 
+    this.el.shadowRoot.getElementById('fileInputField').click();
+  }
+
+  /**
+   * Unable to set a default text value when the prop is marked
+   * as required so generating the default text with this utility.
+   * @returns 
+   */
+  private getButtonText = (): string => {
+    return this.buttonText ? this.buttonText : 'Upload file';
+  };
```

Comment:
If we are defaulting the value anyway, why do we want to mark the prop as required?

Reply:
You're right, the prop does not need to be marked as required (I will update that). But after testing setting a default value on it, I see that we will still need this utility so that it always defaults to "Upload file" if no other text is provided.

Reply:
Resolved here: https://github.com/department-of-veterans-affairs/component-library/pull/507/commits/89adf95016f7a8b180d8cda20c7743e31b611206